### PR TITLE
Remove ConventionProperty

### PR DESCRIPTION
**Pull request by lptr**
Convention mapping has been replaced with providers.

### CODE REVIEW

File: subprojects/core/src/main/java/org/gradle/api/dsl/ConventionProperty.java [hunk 1]
```diff
@@ -1,74 +0,0 @@
...
- * //on a 'test' task point to a wrong folder?
- * </pre>
- *
- * Answer: It will all work fine!
- * <p>
- * Thanks to the 'lazy' evaluation of the convention properties
- * the user can reconfigure the sourceSets anywhere in the gradle script -
- * and still the test.testClassesDir will point to the right folder.
- */
-public class ConventionProperty {}
```
Comment (wolfs) [inline subprojects/core/src/main/java/org/gradle/api/dsl/ConventionProperty.java]: How can we remove this class without going through a deprecation cycle first? Is this class only there for documentation? Could anybody have extended this class?

Comment (lptr) [inline subprojects/core/src/main/java/org/gradle/api/dsl/ConventionProperty.java]: Good point, I removed the Javadoc and made it deprecated. I also deprecated `org.gradle.api.dsl` as there are no other production classes in that package now.

File: subprojects/ide/src/main/java/org/gradle/plugins/ide/idea/model/IdeaProject.java [hunk 2]
```diff
@@ -181,7 +181,7 @@ public String getName() {
     }
 
     /**
-     * A {@link org.gradle.api.dsl.ConventionProperty} that holds modules for the ipr file.
+     * Holds modules for the ipr file.
```
Comment (wolfs) [inline subprojects/ide/src/main/java/org/gradle/plugins/ide/idea/model/IdeaProject.java]: How about "Modules for the ipr file"?

Comment (lptr) [inline subprojects/ide/src/main/java/org/gradle/plugins/ide/idea/model/IdeaProject.java]: Fixed.

**Comment by lptr** [comment]
@wolfs:

> Should we gather all the files we want to remove in some document so we do not forget about this?

When 5.0 comes around we'll check at every `@Deprecated` element.

**Pull request merged**